### ISSUE TITLE
[DEMIS2SORMAS] Handle New Profile: Map Observation.identifier

### ISSUE THREAD

**Issue by FredrikSchaeferVitagroup** [hzi-braunschweig/SORMAS-Project#5713]
### Situation Description

Preriquisite #5539 

The new DEMIS profiles introduce lifecycle management of DEMIS notifications. This means one notification can relate to previously sent notifications. They can, among other references, reference previously trasmitted observations, which are something like pathogen tests in SORMAS terminology. As a user, I want SORMAS to automatically relate such demis notifications to the same pathogen test(s).

### Feature Description

For that, it is necessary to store the [Observation.Identifier](https://simplifier.net/demis/pathogendetectioncvdp) to a pathogen test.
- [ ] Add a externalId String attribute to the TestReport dto and entity.
- [ ] Add this attribute also to the PathogenTest dto and entity. 
- [ ] Show this new attribute for the PathogenTest in the UI when creating or editing pathogenTests.
- [ ] When processing lab messages, always transfer the externalId from the related test report to pathogen test.

### Possible Alternatives

The PathogenTest.typingId can not be used here, or?

### Additional Information

Out of scope for this ticket is to fill the testReport.externalId when converting demis notifications to lab messages in the demis-adapter. This is to be done in [TICKET TO BE CREATED].

**Comment by FredrikSchaeferVitagroup** [comment]
For verifying this issue, check whether the externalId field is always present as specified. Please consider the reasoning [here](https://github.com/hzi-braunschweig/SORMAS-Project/pull/6721#issue-997932288).
To verify the automatic transfer from lab messages, one needs to add an externalId to a test report in the database and process it afterwards.
For further verification, one could also look at edge scenarios (the externalId getting maunally changed while processing a lab message etc.).

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup @AndyBakcsy-she I am trying to test whether fetched lab message that include an Observation.Identifier will result in a pathogen test having a filled externalId field. However, I am struggling to add it to a lab message because I don't know how. Does any of you know how that works or has such a lab message already?

**Comment by FredrikSchaeferVitagroup** [comment]
No, I have not yet uploaded it to a DEMIS-server. In the XML representation, it looks like that:

```
    <entry>
        <fullUrl value="https://demis.rki.de/fhir/Observation/d3539a4c-9093-467a-a3e8-050594afbd59" />
        <resource>
            <Observation>
                <meta>
                    <profile value="https://demis.rki.de/fhir/StructureDefinition/PathogenDetectionCVDP" />
                </meta>
                <identifier>
                    <value value="d3539a4c-9093-467a-a3e8-050594afbd59"/>
                </identifier>
                <status value="final" />
                <category>
                    <coding>
                        <system value="http://terminology.hl7.org/CodeSystem/observation-category" />
                        <code value="laboratory" />
                    </coding>
                </category>
                <code>
                    <coding>
                        <system value="http://loinc.org" />
                        <code value="96741-4" />
                        <display value="SARS-CoV-2 (COVID-19) variant Sequencing Nom (Specimen)" />
                    </coding>
                </code>
                <subject>
                    <reference value="Patient/61ebe359-bfdc-4613-8bf2-c5e300945f0n17" />
                </subject>
                <valueString value="#B.1.1.7" />
                <interpretation>
                    <coding>
                        <system value="http://terminology.hl7.org/CodeSystem/v3-ObservationInterpretation" />
                        <code value="POS" />
                    </coding>
                </interpretation>
                <note>
                    <text value="Ich bin ein optionaler beschreibender Text ..." />
                </note>
                <specimen>
                    <reference value="Specimen/47d61bca-9c53-4422-bb33-3621da084bf4" />
                </specimen>
            </Observation>
        </resource>
    </entry>
```

with `d3539a4c-9093-467a-a3e8-050594afbd59` being the Observation.identifier.value in the encoded demis notificaiton. Maybe that helps you?

We can of course also manually set the TestReport.externalId in the SORMAS db, so I could get test reports that have this filled out. But I guess that is not what you want?

**Comment by kwa20** [comment]
@FredrikSchaeferVitagroup Ok, thats definitely helpful in terms in xml, I'm just not sure how to write this into a JSON file which I would upload to the server. However, I'm also not sure if there is even the possibility that by fetching, the Observation.identifier is not correctly written into the testReport

**Comment by FredrikSchaeferVitagroup** [comment]
> @FredrikSchaeferVitagroup Ok, thats definitely helpful in terms in xml, I'm just not sure how to write this into a JSON file which I would upload to the server. However, I'm also not sure if there is even the possibility that by fetching, the Observation.identifier is not correctly written into the testReport

I think there is. If I'm not mistaken, there can be observations with the id being in the `<fullUrl value=...`, but not in a separate identifier.
